### PR TITLE
SUP-1320: Remove deprecated deletion_protection from pipeline

### PR DESCRIPTION
**Pull request by lizrabuya**
## PR checklist:
- [x] `docs/` updated
- [x] tests updated without deletion_protection
- [x] removed references in`examples/`  
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

File: docs/resources/pipeline.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ resource "buildkite_pipeline" "repo2-release" {
 -   `team` - (Optional) **DEPRECATED** Set team access for the pipeline. Can be specified multiple times for each team.
 -   `tags` - (Optional) A set of tags to be set to the pipeline. For example `["terraform", "provider"]`.
 -   `provider_settings` - (Optional) Source control provider settings for the pipeline. See [Provider Settings Configuration](#provider-settings-configuration) below for details.
--   `deletion_protection` - **DEPRECATED** (Optional) Set to either `true` or `false`. When set to `true`, `destroy` actions on a pipeline will be blocked and fail with a message "Deletion protection is enabled for pipeline: <pipeline name>"
```
Comment (jradtilbrook) [inline docs/resources/pipeline.md:110]: This needs line 56 removed as well

**Review by jradtilbrook — CHANGES_REQUESTED** [review]

File: docs/resources/pipeline.md [hunk 2]
```diff
@@ -53,7 +53,9 @@ resource "buildkite_pipeline" "test_new" {
 [Lifecycles](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle) allow you to set general rules
 for resources, including the prevention of destruction of a resource. In the above example, the `destroy` command on the `Testing TImeouts` pipeline will fail.
 
-Lifecycles will replace the deprecated `deletion_protection` in `v1` of the provider in favour of `lifecycle` rules.
+#### Deprecation notice
+`archive_on_delete` has been deprecated and moved to provider configuration instead. The usage through provider
+configuration makes more sense and will apply to all pipelines managed through terraform. Please see `archive_pipeline_on_delete`.
```
Comment (jradtilbrook) [inline docs/resources/pipeline.md:58]: I think this is a bad merge. This should have been removed as well

**Review by jradtilbrook — CHANGES_REQUESTED** [review]

File: docs/resources/pipeline.md [hunk 3]
```diff
@@ -53,7 +53,9 @@ resource "buildkite_pipeline" "test_new" {
...
```
Comment (mcncl) [inline docs/resources/pipeline.md:59]: Probably needs deleting?

**Review by mcncl — CHANGES_REQUESTED** [review]

File: docs/resources/pipeline.md (hunk 3 above)
Comment (mcncl) [reply docs/resources/pipeline.md:59]: Resolving as same as above.

**Review by mcncl — APPROVED** [review]

**Review by james2791 — APPROVED** [review]

**Review by jradtilbrook — APPROVED** [review]

**Pull request merged**